converter/converter.go

```diff
@@ -28,10 +29,16 @@ const (
 // because of mismatched functionality, an error is returned with no resulting
 // config. If the conversion completed successfully but generated warnings, an
 // error is returned alongside the resulting config.
-func Convert(in []byte, kind Input) ([]byte, error) {
+func Convert(in []byte, kind Input) ([]byte, diag.Diagnostics) {
```

Comment:
Personally, I'd prefer creating a new error type for this instead of trying to reuse river/diag; it doesn't feel like a natural fine since there's never any file name or line information.